app/src/main/res/layout/fragment_grid_memes.xml

```diff
@@ -6,36 +6,22 @@
     android:layout_height="match_parent"
     tools:context=".ui.gridmemes.GridMemesFragment">
 
-    <TextView
-        android:id="@+id/activity_main_call_service_text_view"
-        android:layout_width="wrap_content"
+    <ProgressBar
+        android:id="@+id/fragment_grid_memes_progress_bar"
+        style="?android:attr/progressBarStyle"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:text="@string/push_to_call_service"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        android:visibility="gone"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent" />
 
-    <Button
-        android:id="@+id/activity_main_call_service_button"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_marginTop="@dimen/margin_top_service_button"
-        android:text="@string/call"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
-
-    <TextView
-        android:id="@+id/activity_main_first_meme_name"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_marginTop="@dimen/margin_top_first_meme_name"
-        android:text="@string/first_meme_id"
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/fragment_grid_memes_recycler_view"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
         app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
```

Comment:
if you use
app:layout_constraintEnd_toEndOf="parent"
app:layout_constraintStart_toStartOf="parent"
android:layout_width="0dp"
you can replace irt for
android:layout_width="match_parent"

Reply:
done, thanks